src/avram/query_builder.cr

```diff
@@ -125,8 +127,16 @@ class Avram::QueryBuilder
     self
   end
 
-  private def distinct?
-    @distinct || @distinct_on
+  def distinct?
+    @distinct || has_distinct_on?
+  end
+
+  def has_distinct_on?
```

Comment:
I’m not sure I understand this method. I think it was clearer when using a conditional directly with `@distinct_on` But maybe I’m misunderstanding

Reply:
I needed this separate because there may be times where `distinct?` would return true, but you don't actually have `distinct_on`. So I need to know if you have both separately. This way it's like "merge distinct if you have distinct", but then "merge distinct_on if you have distinct_on".